pkg/fleets/controller.go

```diff
@@ -169,7 +168,7 @@ func (c *Controller) syncFleet(key string) error {
 		return errors.Wrapf(err, "error retrieving fleet %s from namespace %s", name, namespace)
 	}
 
-	list, err := c.listGameServerSets(fleet)
+	list, err := ListGameServerSetsByFleetOwner(c.gameServerSetLister, fleet)
```

Comment:
@enocom would love to get your opinion on this refactor ([implementation here](https://github.com/GoogleCloudPlatform/agones/pull/193/files#diff-91029c2ecd86ab36712b39724f333a94R28))

There were several places I needed a list type function like this, and wasn't quite sure the best way to do it.

Reply:
See my response below.